src/command/defaults/ClearCommand.php

```diff
@@ -0,0 +1,178 @@
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\command\defaults;
+
+use InvalidArgumentException;
+use pocketmine\command\CommandSender;
+use pocketmine\command\utils\InvalidCommandSyntaxException;
+use pocketmine\item\LegacyStringToItemParser;
+use pocketmine\lang\TranslationContainer;
+use pocketmine\player\Player;
+use pocketmine\utils\TextFormat;
+use function array_merge;
+use function count;
+use function is_numeric;
+
+class ClearCommand extends VanillaCommand{
+
+	public function __construct(string $name){
+		parent::__construct(
+			$name,
+			"%pocketmine.command.clear.description",
+			"%pocketmine.command.clear.usage"
+		);
+		$this->setPermission("pocketmine.command.clear.self;pocketmine.command.clear.other");
+	}
+
+	public function execute(CommandSender $sender, string $commandLabel, array $args){
+		if(!$this->testPermission($sender)){
+			return true;
+		}
+
+		if(count($args) > 3){
+			throw new InvalidCommandSyntaxException();
+		}
+
+		$target = null;
+		if($sender instanceof Player){
+			if(!$sender->hasPermission("pocketmine.command.clear.self")){
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.generic.permission"));
+				return true;
+			}
+
+			$target = $sender;
+		}else{
+			if(count($args) < 1){
+				throw new InvalidCommandSyntaxException();
+			}
+		}
+
+		if(isset($args[0])){
+			if(!$sender->hasPermission("pocketmine.command.clear.other")){
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.generic.permission"));
+				return true;
+			}
+
+			$target = $sender->getServer()->getPlayer($args[0]);
+		}
+
+		if($target === null){
+			$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.generic.player.notFound"));
+			return true;
+		}
+
+		$item = null;
+		if(isset($args[1])){
+			try{
+				$item = LegacyStringToItemParser::getInstance()->parse($args[1]);
+			}catch(InvalidArgumentException $e){
+				//vanilla checks this at argument parsing layer, can't come up with a better alternative
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.give.item.notFound", [$args[1]]));
+				return true;
+			}
+		}
+
+		$maxCount = -1;
+		if(isset($args[2])){
+			if(!is_numeric($args[2])){
+				$maxCount = -1;
+			}
+
+			if($item !== null){
+				$item->setCount($maxCount = $this->getInteger($sender, $args[2], 0));
+			}
+		}
+
+		//checking players inventory for all the items matching the criteria
+		if($item !== null and $item->getMeta() === 0 and $maxCount === 0){
```

Comment:
why is there a `getMeta()` check here?

Reply:
Vanilla `clear` command does more than just simply clearing the inventory of the player. Vanilla syntax is, 
```clear [player] [item name] [data] [max count]```
Where item being the item name and data being the damage/meta. If the meta and the max count are set to zero then it reports back all the items matching the items specified in the players inventory. 
More info [here](https://www.digminecraft.com/game_commands/clear_command.php) or see the attached screenshot.

![image](https://user-images.githubusercontent.com/32965703/95328052-68cce900-08be-11eb-9028-3a30bfe57acf.png)

Reply:
This seems out of character for a `/clear` command...

Reply:
True but that's how its done in vanilla... Just to be sure, I checked this in-game as well.

Reply:
OK, so /clear has some behaviour that's not relevant to clearing when using count zero. Fine. But this still doesn't explain the metadata zero check.

Reply:
EX: `/clear steve bucket 8 0` has the same behaviour as `/clear steve bucket 0 0`.

Reply:
Ah true, forgot about that.